packages/backend/src/open_payments/payment/outgoing/lifecycle.ts

```diff
@@ -158,6 +164,295 @@ function getAdjustedAmounts(
   return { maxDebitAmount, maxReceiveAmount }
 }
 
+/**
+ * Compares the final settled amounts with the amounts on hold
+ * and inserts a new OutgoingPaymentGrantSpentAmount record if needed.
+ */
+async function handleGrantSpentAmounts(
```

Comment:
I realized we also need to call this whenever we call `cancelOutgoingPayment` as well

Reply:
added